### PR TITLE
[v16] Expose client tools auto update for find endpoint

### PR DESCRIPTION
**Pull request by vapopov**
Backport https://github.com/gravitational/teleport/pull/46785 and https://github.com/gravitational/teleport/pull/47532 to branch/v16

### CODE REVIEW

**Comment by vapopov** [comment]
@rosstimothy I've cherry picked recent changes in proto structure here as well

File: api/client/webclient/webclient.go [hunk 1]
```diff
@@ -328,6 +330,14 @@ type ProxySettings struct {
 	TLSRoutingEnabled bool `json:"tls_routing_enabled"`
 }
 
+// AutoUpdateSettings contains information about the auto update requirements.
+type AutoUpdateSettings struct {
+	// ToolsVersion defines the version of {tsh, tctl} for client auto update.
+	ToolsVersion string `json:"tools_version"`
+	// ToolsMode defines mode client auto update feature `enabled|disabled`.
+	ToolsMode string `json:"tools_mode"`
```
Comment (sclevine) [inline api/client/webclient/webclient.go:338]: @vapopov I think this should be `tools_auto_update`?

https://github.com/gravitational/teleport/blob/master/api/client/webclient/webclient.go#L363

**Comment by sclevine** [comment]
Should this PR be closed in favor of https://github.com/gravitational/teleport/pull/48645?

**Comment by vapopov** [comment]
closing in favor https://github.com/gravitational/teleport/pull/48645

**Pull request closed without merging**